packages/react-form/src/hooks/field/field.ts

```diff
@@ -105,10 +110,14 @@ export function useField<Value = string>(
   input: FieldConfig<Value> | Value,
   dependencies: unknown[] = [],
 ): Field<Value> {
-  const {value, validates} = normalizeFieldConfig(input);
+  const {
+    value,
+    validates,
+    dirtyStateComparator = defaultDirtyComparator,
```

Comment:
i think i would consider pushing this default comparator assignment all the way down to `makeFieldReducer`. let the optional value reach as far as it can and assign its default once the value becomes necessary.